hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -807,18 +804,24 @@ def conclude_test(
         Note that this is called after ``freeze`` has completed.
         """
 
-    def draw_bits(self, n_bits: int, *, forced: bool, value: int) -> None:
-        """Called when ``draw_bits`` is called on on the
-        observed ``ConjectureData``.
-        * ``n_bits`` is the number of bits drawn.
-        *  ``forced`` is True if the corresponding
-           draw was forced or ``False`` otherwise.
-        * ``value`` is the result that ``draw_bits`` returned.
-        """
-
     def kill_branch(self) -> None:
         """Mark this part of the tree as not worth re-exploring."""
 
+    def draw_integer(self, value: int, forced: bool, *, kwargs: dict) -> None:
+        pass
+
+    def draw_float(self, value: float, forced: bool, *, kwargs: dict) -> None:
+        pass
+
+    def draw_string(self, value: str, forced: bool, *, kwargs: dict) -> None:
+        pass
+
+    def draw_bytes(self, value: bytes, forced: bool, *, kwargs: dict) -> None:
+        pass
+
+    def draw_boolean(self, value: bool, forced: bool, *, kwargs: dict) -> None:
+        pass
+
```

Comment:
we could move this to a single `def draw(self, ir_type, ...)` method, but it's sort of half dozen of one 6 of the other. Writing them out explicitly (shown here) might be better for facilitating strong typing, but I don't have a strong opinion.

Reply:
I'd favor writing out explicit `(self, value: ..., / was_forced: bool, *, min_value, ...)` - it's somewhat more verbose, but much harder to screw up later, and we can get type-checker support for this at some point (#3074).